docs/Reference/Hooks.md

```diff
@@ -416,6 +418,31 @@ fastify.addHook('onReady', async function () {
 })
 ```
 
+### onListen
+
+Triggered when the server starts listening for requests. The hooks run one 
+after another. If a hook function causes an error, it is logged and 
+ignored, allowing the queue of hooks to continue. Hook functions accept one
+argument: a callback, `done`, to be invoked after the hook function is
+complete. Hook functions are invoked with `this` bound to the associated
+Fastify instance.
+
+This is an alternative to `fastify.server.on('listening', () => {})`.
+
+```js
+// callback style
+fastify.addHook('onListen', function (done) {
+  // Some code
+  const err = null;
+  done(err)
+})
+
+// or async/await style
+fastify.addHook('onListen', async function () {
+  // Some async code
+})
+```
+
```

Comment:
```suggestion

> **Note**  
> This hook will not run when the server is started using `fastify.inject()` or `fastify.ready()`

```

Reply:
Kind of confusing. Ready ok, as ready happens before listening. But i can still inject() after i called listen().

Reply:
What about:

```suggestion

**Note**  
This hook will only run when the server is started using `fastify.listen()`. Using only `fastify.inject()` or `fastify.ready()` won't trigger the hook.
```

?

Reply:
@Fdawgs